tests/main/snap-quota/task.yaml

```diff
@@ -0,0 +1,48 @@
+summary: Basic test for quota-related snap commands.
+
+details: |
+  Basic test for snap quota, remove-quota and quotas commands.
+
+prepare: |
+  snap install hello-world
+  snap install test-snapd-tools
+
+execute: |
+  echo "Creating top-level quota groups (no snaps)"
+  snap quota group-top1 --max-memory 4MB
+  snap quota group-top2 --max-memory 500MB
+
+  echo "Creating groups with snaps in them"
+  snap quota group-one --parent group-top1 --max-memory 1MB hello-world
+  snap quota group-two --parent group-top1 --max-memory 2MB test-snapd-tools
+
+  echo "Creating some more nested empty quota groups"
+  snap quota group-sub-one --parent group-one
+  snap quota group-sub-sub-one --parent group-sub-one
+
+  echo "Trying to add snap to more than one group fails"
+  snap quota group-bad --max-memory 1MB hello-world 2>&1 | MATCH "XXX"
```

Comment:
the error here is 
```
MATCH 'error: cannot create or update quota group: cannot use snap "hello-world" in group "group-bad": snap already in quota group "group-one"'
```
which highlights we need to cleanup this part too later it's a bit repetitive :)

Reply:
Yes yes, not surprising/expected, as I said this is a draft and I entirely hand-crafted this test without any way to run it. Failures totally expected. Will work on it and error messages.